panoptes_aggregation/tests/panoptes_tests/test_userify.py

```diff
@@ -0,0 +1,134 @@
+from importlib import import_module
+
+from unittest.mock import MagicMock, PropertyMock, Mock
+from nose.tools import assert_equals
+panoptes = import_module('panoptes_aggregation.panoptes', __name__).panoptes_testing
+from panoptes_client import Panoptes, User
+
+
+def build_mock_user(**kwargs):
+    mock_user = MagicMock()
+    for key, value in kwargs.items():
+        setattr(type(mock_user), key, PropertyMock(return_value=value, create=True))
+    return mock_user
+
+
+def test_userify():
+    pass
+
+
+def test_unique():
+    # removes duplicate elements
+    assert_equals(panoptes._unique([]),[])
+    assert_equals(panoptes._unique([3]), [3])
+    assert_equals(panoptes._unique([3, 3]), [3])
+    assert_equals(panoptes._unique([3, 3, 3]), [3])
+    assert_equals(panoptes._unique([3,4]), [3,4])
+    assert_equals(panoptes._unique([1, 1, 2, 3]), [1, 2, 3])
+    assert_equals(panoptes._unique([1, 2, 3, 1]), [1, 2, 3])
+
+
+def test_flatten():
+    # makes any combination of lists and not-lists into a single flat list
+    assert_equals(list(panoptes._flatten([])), [])
+    assert_equals(list(panoptes._flatten([1])), [1])
+    assert_equals(list(panoptes._flatten([1, 2])), [1, 2])
+    assert_equals(list(panoptes._flatten([1, [2]])), [1, 2])
+    assert_equals(list(panoptes._flatten([[1], 2])), [1, 2])
+    assert_equals(list(panoptes._flatten([[1], [2]])), [1, 2])
+    assert_equals(list(panoptes._flatten([1, [2], 3])), [1, 2, 3])
+    assert_equals(list(panoptes._flatten([1, [2], 2])), [1, 2, 2])
+    assert_equals(list(panoptes._flatten([[1], 2, []])), [1, 2])
+    assert_equals(list(panoptes._flatten([[1], [], 2])), [1, 2])
+
+
+def test_discover_fields():
+    # can find simple list of properties
+    assert_equals(panoptes._discover_fields({}), [])
+    assert_equals(panoptes._discover_fields({'f1': 'v1'}), ['f1'])
+    assert_equals(panoptes._discover_fields({'f1': 'v1', 'f2': 'v2'}), ['f1', 'f2'])
+    assert_equals(panoptes._discover_fields({'f1': 'v1', 'destination': 'd1', 'f2': 'v2'}), ['f1', 'f2'])
```

Comment:
Python `dict`s are unordered and half the time this will return `['f2, 'f1']` and the test will fail.  I have been using the [assertCountEqual](https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.assertCountEqual) for this, it does not care about the order (although I have not tried using this outside of a `unittest.TestCase` class instance).

Reply:
good call--and nose wraps `assertCountEqual` in `nose.tools.assert_count_equal` too!

Reply:
done